python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -333,7 +333,8 @@ def set_object_dtypes_from_pa_schema(df, schema):
     if schema:
         for col_name, col in df._data.items():
             if col_name in schema.names and isinstance(
-                col, cudf.core.column.StringColumn
+                cudf_dtype_from_pa_type(schema.field(col_name).type),
```

Comment:
`cudf_dtype_from_pa_type` is expected to return a `dtype`, but doing an `isinstance` check on `dtype` vs `cudf.core.column.StringColumn` will always return False no?

Reply:
Ah - My mistake. I guess we will just need an explicit check that the cudf dtype will not be one of the "non-castable" dtypes (i.e. "list" or "struct")?

Reply:
Can we not directly have `cudf.core.column.ListColumn`/`cudf.core.column.StructColumn` instance checks?

Reply:
>Can we not directly have cudf.core.column.ListColumn/cudf.core.column.StructColumn instance checks?

Yes. That's fair :)

Reply:
>>Can we not directly have cudf.core.column.ListColumn/cudf.core.column.StructColumn instance checks?
>
>Yes. That's fair :)

On second thought, I'm not actually sure how to make that instance check work.  The problem we are dealing with here is that cudf does not allow us to "correct" the dask `_meta` by casting the columns of `_meta` to the appropriate arrow->cudf dtype conversion when the "correct" dtypes are "list" or "struct" (and the _meta column dtypes are still "object").  

The approach taken by `set_object_dtypes_from_pa_schema` is to cast existing `_meta` columns to the "correct" dtype, but cudf does not allow casting to "list" or "struct" dtpes.  Therefore, this "fix" is really just giving up on getting the metadata right when one of these dtypes is expected.

Reply:
okay - Latest commit hopefully addresses the messiness here a bit better :)

Reply:
Thanks @rjzamora !